src/osd/strconv.cpp

```diff
@@ -19,23 +19,28 @@
 //  astring_from_utf8
 //============================================================
 
-CHAR *astring_from_utf8(const char *utf8string)
+std::string &astring_from_utf8(std::string &dst, const char *s)
 {
-	WCHAR *wstring;
-	int char_count;
-	CHAR *result;
-
 	// convert MAME string (UTF-8) to UTF-16
-	char_count = MultiByteToWideChar(CP_UTF8, 0, utf8string, -1, nullptr, 0);
-	wstring = (WCHAR *)alloca(char_count * sizeof(*wstring));
-	MultiByteToWideChar(CP_UTF8, 0, utf8string, -1, wstring, char_count);
+	std::basic_string<WCHAR> wstring = wstring_from_utf8(s);
 
 	// convert UTF-16 to "ANSI code page" string
-	char_count = WideCharToMultiByte(CP_ACP, 0, wstring, -1, nullptr, 0, nullptr, nullptr);
-	result = (CHAR *)osd_malloc_array(char_count * sizeof(*result));
-	if (result != nullptr)
-		WideCharToMultiByte(CP_ACP, 0, wstring, -1, result, char_count, nullptr, nullptr);
+	int char_count = WideCharToMultiByte(CP_ACP, 0, wstring.c_str(), -1, nullptr, 0, nullptr, nullptr);
```

Comment:
You have a std::string as input.  It knows its length.  You be passing that rather than -1 as cchWideChar to avoid an expensive scan.  Same goes for the call below.